### DISCUSSION TITLE
How from HeaderWidgets=>getViewData method get parent resource $record  ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#6551] [Help]
On site with filament-widget I added Item resource widget with command :

php artisan make:filament-widget ItemStats --resource=ItemResource

  In app/Filament/Resources/ItemResource/Pages/EditItem.php :
```
    protected function getHeaderWidgets(): array
    {
        return [
            ItemStats::class,
        ];
    }

```

In file app/Filament/Resources/ItemResource/Widgets/ItemStats.php I added method to get data:

```
    protected function getViewData(): array
    {
        $itemStats = ItemStat::
           ...

```
But how can I in getViewDatamethod to get Item resource $record ? I need it to get related data from ItemStat model ?

```
"filament/filament": "^2.17.34",
"laravel/framework": "^9.52.5",

```

Thanks in advance!

**Comment by danharrin** [comment]
`$this->record`

**Reply by PetroGromovo** [reply]
I got error :
`Property [$record] not found on component:
`
As my widget extends vendor/filament/filament/src/Widgets/Widget.php I did not find any $record defined in it :


```
<?php

namespace Filament\Widgets;

use Illuminate\Contracts\View\View;
use Livewire\Component;

class Widget extends Component
{
    protected static ?int $sort = null;

    protected static string $view;

    protected int | string | array $columnSpan = 1;

    public static function canView(): bool
    {
        return true;
    }

    public static function getSort(): int
    {
        return static::$sort ?? -1;
    }

    public function getColumnSpan(): int | string | array
    {
        return $this->columnSpan;
    }

    protected function getViewData(): array
    {
        return [];
    }

    public function render(): View
    {
        return view(static::$view, $this->getViewData());
    }
}

```

**Reply by danharrin** [reply]
You don't need `getViewData()`. `$record` will be automatically exposed to the view, and `$this->record` in the class. You need to add the property:

https://filamentphp.com/docs/2.x/admin/resources/widgets#accessing-the-current-record
